In [9]:

import plotly.express as px


In [10]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [11]:
#%sql postgresql://postgres:Letsdoit!@localhost:5432/Eskom_db
%sql postgresql://postgres:postgres@localhost:5430/eskom_db

'Connected: postgres@eskom_db'

In [14]:
%%sql 
DROP TABLE IF EXISTS eskom_data;
CREATE TABLE  eskom_data(
    date_time_hour_beginning DATE,
    original_res_forecast_before_lockdown NUMERIC,
    residual_forecast  NUMERIC,	
    rsa_contracted_forecast	 NUMERIC,
    dispatchable_generation	NUMERIC ,
    residual_demand NUMERIC,
    rsa_contracted_demand  NUMERIC,
    international_exports  NUMERIC,
    international_imports  NUMERIC,
    thermal_generation  NUMERIC,
    nuclear_generation  NUMERIC,
    eskom_gas_generation  NUMERIC,	
    eskom_ocgt_generation  NUMERIC,	
    hydro_water_generation  NUMERIC,
    pumped_water_generation	 NUMERIC,
    ils_usage  NUMERIC,
    manual_load_reduction NUMERIC,
    ios_excl_ils_and_mlr  NUMERIC,
    dispatchable_ipp_ocgt  NUMERIC,
    eskom_gas_sco  NUMERIC,
    eskom_ocgt_sco  NUMERIC,
    hydro_water_sco	 NUMERIC,
    pumped_water_sco  NUMERIC,
    pumping	 NUMERIC,
    wind  NUMERIC,
    pv  NUMERIC,
    csp  NUMERIC,
    other_re  NUMERIC,	
    total_re  NUMERIC,	
    wind_installed_capacity	 NUMERIC,
    pv_installed_capacity  NUMERIC,	
    csp_installed_capacity  NUMERIC,
    other_re_installed_capacity	 NUMERIC,
    total_re_installed_capacity	 NUMERIC,
    installed_eskom_capacity  NUMERIC,
    total_pclf  NUMERIC,
    total_uclf  NUMERIC,
    total_oclf  NUMERIC,
    total_uclf_and_oclf	 NUMERIC,
    non_comm_sentout  NUMERIC,	
    drakensberg_gen_unit_hours  NUMERIC,
    palmiet_gen_unit_hours  NUMERIC,	
    ingula_gen_unit_hours NUMERIC
    
);


    

 * postgresql://postgres:***@localhost:5430/eskom_db
Done.
Done.


[]

In [13]:
#FROM 'C:\Users\Public\ESK2033_clean.csv'

In [19]:
%%sql 
COPY eskom_data(date_time_hour_beginning,original_res_forecast_before_lockdown,residual_forecast,rsa_contracted_forecast,dispatchable_generation,residual_demand,rsa_contracted_demand,international_exports,international_imports,thermal_generation,nuclear_generation,eskom_gas_generation,eskom_ocgt_generation,hydro_water_generation,pumped_water_generation,ils_usage,manual_load_reduction,ios_excl_ils_and_mlr,dispatchable_ipp_ocgt,eskom_gas_sco,eskom_ocgt_sco,hydro_water_sco,pumped_water_sco,pumping,wind,pv,csp,other_re,total_re,wind_installed_capacity,pv_installed_capacity,csp_installed_capacity,other_re_installed_capacity,total_re_installed_capacity,installed_eskom_capacity,total_pclf,total_uclf,total_oclf,total_uclf_and_oclf,non_comm_sentout,drakensberg_gen_unit_hours,palmiet_gen_unit_hours,ingula_gen_unit_hours)
FROM '/Users/ds_learner_14/Documents/Eskom_Data_Analysis/ESK2033.csv'
DELIMITER ','
CSV HEADER
;

 * postgresql://postgres:***@localhost:5430/eskom_db
37704 rows affected.


[]

In [20]:
%%sql 
DROP TABLE IF EXISTS eskom_clean;
CREATE TABLE eskom_clean 
AS 
SELECT
    EXTRACT(YEAR FROM date_time_hour_beginning) as year,
    EXTRACT(MONTH FROM date_time_hour_beginning) as month,
    EXTRACT(DAY FROM date_time_hour_beginning) as day,
    date_time_hour_beginning,
    original_res_forecast_before_lockdown,
    residual_forecast,
    rsa_contracted_forecast,
    dispatchable_generation,
    residual_demand,
    rsa_contracted_demand,
    international_exports,
    international_imports,
    thermal_generation,
    nuclear_generation,
    eskom_gas_generation,
    eskom_ocgt_generation,
    hydro_water_generation,
    pumped_water_generation,
    ils_usage,
    manual_load_reduction,
    ios_excl_ils_and_mlr,
    dispatchable_ipp_ocgt,
    eskom_gas_sco,
    eskom_ocgt_sco,
    hydro_water_sco,
    pumped_water_sco,
    pumping,
    wind,
    pv,
    csp,
    other_re,
    total_re,
    wind_installed_capacity,
    pv_installed_capacity,
    csp_installed_capacity,
    other_re_installed_capacity,
    total_re_installed_capacity,
    installed_eskom_capacity,
    total_pclf,
    total_uclf,
    total_oclf,
    total_uclf_and_oclf,
    non_comm_sentout,
    drakensberg_gen_unit_hours,
    palmiet_gen_unit_hours,
    ingula_gen_unit_hours

FROM eskom_data;


 * postgresql://postgres:***@localhost:5430/eskom_db
Done.
75408 rows affected.


[]

# Numbers of rows and columns
 count rows=42824 estimated before clean after 37704 removedforecast dates added average for original res before lockdown 
count cols=41


look at our variables


In [21]:
all = %sql SELECT * FROM eskom_clean LIMIT 5 
all

 * postgresql://postgres:***@localhost:5430/eskom_db
5 rows affected.


[(Decimal('2018'), Decimal('4'), Decimal('1'), datetime.date(2018, 4, 1), Decimal('27517.5'), Decimal('19904.967'), Decimal('20367.066'), Decimal('20237'), Decimal('20237'), Decimal('20722.058'), Decimal('1215.902'), Decimal('1120'), Decimal('19444'), Decimal('931'), Decimal('0'), Decimal('0'), Decimal('360'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('-2'), Decimal('-2'), Decimal('0'), Decimal('-1614'), Decimal('454.964'), Decimal('0'), Decimal('17.555'), Decimal('12.539'), Decimal('485.058'), Decimal('2077.72'), Decimal('1474.19'), Decimal('300'), Decimal('21.78'), Decimal('3873.69'), Decimal('44546'), Decimal('3987.472'), Decimal('8028.71'), Decimal('275.907'), Decimal('8304'), Decimal('617'), Decimal('383'), Decimal('81.8'), Decimal('36.9'), Decimal('30.41')),
 (Decimal('2018'), Decimal('4'), Decimal('1'), datetime.date(2018, 4, 1), Decimal('27517.5'), Decimal('19553.899'), Decimal('19988.733'), Decimal('19744'), Decimal('19744'), Decimal('20188.493'), Decimal('1203.474'), Decimal('1106'), Decimal('19297'), Decimal('930'), Decimal('0'), Decimal('0'), Decimal('360'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('-2'), Decimal('-2'), Decimal('0'), Decimal('-1945'), Decimal('392.958'), Decimal('0'), Decimal('39.131'), Decimal('12.404'), Decimal('444.493'), Decimal('2077.72'), Decimal('1474.19'), Decimal('300'), Decimal('21.78'), Decimal('3873.69'), Decimal('44546'), Decimal('3987.472'), Decimal('7727.302'), Decimal('244.907'), Decimal('7972'), Decimal('209'), Decimal('388'), Decimal('83'), Decimal('38.5'), Decimal('32.85')),
 (Decimal('2018'), Decimal('4'), Decimal('1'), datetime.date(2018, 4, 1), Decimal('27517.5'), Decimal('19314.284'), Decimal('19731.239'), Decimal('19631'), Decimal('19631'), Decimal('20019.603'), Decimal('1177.571'), Decimal('1117'), Decimal('19165'), Decimal('931'), Decimal('0'), Decimal('0'), Decimal('360'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('-2'), Decimal('-2'), Decimal('0'), Decimal('-1938'), Decimal('337.358'), Decimal('0'), Decimal('39.163'), Decimal('12.082'), Decimal('388.603'), Decimal('2077.72'), Decimal('1474.19'), Decimal('300'), Decimal('21.78'), Decimal('3873.69'), Decimal('44546'), Decimal('3987.472'), Decimal('7704.704'), Decimal('193.727'), Decimal('7898'), Decimal('431'), Decimal('388'), Decimal('83.8'), Decimal('40.3'), Decimal('35.6')),
 (Decimal('2018'), Decimal('4'), Decimal('1'), datetime.date(2018, 4, 1), Decimal('27517.5'), Decimal('19342.679'), Decimal('19753.554'), Decimal('19731'), Decimal('19731'), Decimal('20079.454'), Decimal('1184.312'), Decimal('1118'), Decimal('19279'), Decimal('930'), Decimal('0'), Decimal('0'), Decimal('360'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('-2'), Decimal('-2'), Decimal('0'), Decimal('-1952'), Decimal('297.234'), Decimal('0'), Decimal('38.942'), Decimal('12.278'), Decimal('348.454'), Decimal('2077.72'), Decimal('1474.19'), Decimal('300'), Decimal('21.78'), Decimal('3873.69'), Decimal('44546'), Decimal('3990.072'), Decimal('7702.868'), Decimal('187'), Decimal('7889'), Decimal('868'), Decimal('389'), Decimal('85'), Decimal('42'), Decimal('37.76')),
 (Decimal('2018'), Decimal('4'), Decimal('1'), datetime.date(2018, 4, 1), Decimal('27517.5'), Decimal('19538.89'), Decimal('19988.365'), Decimal('19890'), Decimal('19890'), Decimal('20237.49'), Decimal('1197.271'), Decimal('1108'), Decimal('19369'), Decimal('930'), Decimal('0'), Decimal('0'), Decimal('360'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('0'), Decimal('-2'), Decimal('-2'), Decimal('0'), Decimal('-1873'), Decimal('309.978'), Decimal('0'), Decimal('25.024'), Decimal('12.488'), Decimal('347.49'), Decimal('2077.72'), Decimal('1474.19'), Decimal('300'), Decimal('21.78'), Decimal('3873.69'), Decimal('44546'), Decimal('3990.472'), Decimal('7685.115'), Decimal('187'), Decimal('7872'), Decimal('115'), Decimal('385'), Decimal('85.

# comparing

ESKOM GAS SCO

ESKOM OCGT

HYDRO WATER

PUMPED WATER

In [40]:
avg_sco = %sql SELECT year AS "year",AVG(eskom_gas_sco) AS "eskom_gas",AVG(eskom_ocgt_sco) AS "eskom_ocgt",AVG(hydro_water_sco) AS "hydro_water",AVG(pumped_water_sco) AS "pumped_water" FROM eskom_clean WHERE year >= '2018' and year <= '2022' GROUP BY year ORDER BY year
avg_sco

 * postgresql://postgres:***@localhost:5430/eskom_db
5 rows affected.


[(Decimal('2018'), Decimal('-1.9192424242424242'), Decimal('-3.0245469696969697'), Decimal('0E-24'), Decimal('-719.0386363636363636')),
 (Decimal('2019'), Decimal('-1.9611872146118721'), Decimal('-2.2163778538812785'), Decimal('0E-24'), Decimal('-741.4743150684931507')),
 (Decimal('2020'), Decimal('-1.5865209471766849'), Decimal('-3.0372490892531876'), Decimal('0E-24'), Decimal('-779.4633424408014572')),
 (Decimal('2021'), Decimal('-1.8659817351598174'), Decimal('-2.3971826484018265'), Decimal('0E-24'), Decimal('-756.7438356164383562')),
 (Decimal('2022'), Decimal('-1.7543750000000000'), Decimal('-3.9935889583333333'), Decimal('0E-20'), Decimal('-688.0275000000000000'))]

## Which energy uses the most gas to produce electricity?

In [37]:
fig = px.line(avg_sco, x="year", y=["eskom_gas","eskom_ocgt"],title="Average of eskom gas and eskom ocgt")
fig.show()

In [46]:
fig1 = px.line(avg_sco, x="year", y =["hydro_water", "pumped_water"],title="Average of hydro water and pumped water")
fig1.show()